src/kemal/handler.cr

```diff
@@ -15,6 +15,7 @@ class Kemal::Handler < HTTP::Handler
 
   def add_route(method, path, &handler : Kemal::Context -> _)
     @routes << Route.new(method, path, &handler)
+    @routes << Route.new("HEAD", path, &handler) if method == "GET"
```

Comment:
I checked Sinatra's codebase as well after this pull request and they are also doing exactly the same thing i.e. defining `HEAD` routes automatically for each defined `GET` route. For reference please see https://github.com/sinatra/sinatra/blob/master/lib/sinatra/base.rb#L1368